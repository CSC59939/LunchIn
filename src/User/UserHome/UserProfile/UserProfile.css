.user-profile-content {
  display: flex;
  flex-direction: row;
  width: 60%;
  margin: 0 auto;
  justify-content: center !important;
}

.input-paper {
  margin-top: 5px;
  padding: 10px !important;
  text-align: center;
  border-radius: 25px !important;
}

.bar {
  margin-top: 5px;
  border-radius: 20px !important;
}
