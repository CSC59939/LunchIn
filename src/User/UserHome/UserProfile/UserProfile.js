import React, { Component } from "react";
import "./UserProfile.css";
import { firebase } from "../../../Config";
import {
  Button,
  Card,
  CardContent,
  Divider,
  Snackbar,
  TextField,
  Typography,
  Tabs,
  Tab,
  AppBar
} from "@material-ui/core";
import SwipeableViews from "react-swipeable-views";
import Back from "@material-ui/icons/ArrowBack";

function TabContainer({ children, dir }) {
  return (
    <Typography component="div" dir={dir}>
      {children}
    </Typography>
  );
}

class UserProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      email: "",
      currentPassword: "",
      newPassword: "",
      address: "",
      newAddress: "",
      processing: false,
      notify: false,
      notifyMsg: "",
      value: 0
    };
    this.submitNewAddress = this.submitNewAddress.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleChangeTab = this.handleChangeTab.bind(this);
    this.handleChangeIndex = this.handleChangeIndex.bind(this);
    this.onChangePasswordPress = this.onChangePasswordPress.bind(this);
    this.reauthenticate = this.reauthenticate.bind(this);
  }
  componentDidMount() {
    this.fetchInitialData();
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  handleChangeTab = (event, value) => {
    this.setState({ value });
  };

  handleChangeIndex = index => {
    this.setState({ value: index });
  };

  fetchInitialData() {
    firebase.auth().onAuthStateChanged(user => {
      if (user) {
        firebase
          .database()
          .ref(`Users/${user.uid}`)
          .once(
            "value",
            snapshot => {
              if (snapshot.val() != null) {
                this.setState({
                  name: snapshot.val().name,
                  address: snapshot.val().address
                });
              }
            },
            error => {
              this.setState({
                processing: false,
                notify: true,
                notifyMsg: error.message
              });
            }
          );
      }
    });
  }

  submitNewAddress(e) {
    e.preventDefault();
    if (this.state.address) {
      this.setState({
        address: this.state.address,
        processing: true
      });
      const addressRef = firebase
        .database()
        .ref(`Users/${firebase.auth().currentUser.uid}/address`);
      addressRef
        .set(this.state.address)
        .then(() => {
          this.setState({
            processing: false,
            notify: true,
            notifyMsg: "Email update successfully!"
          });
        })
        .catch(error => {
          this.setState({
            notify: true,
            notifyMsg: error.message,
            processing: false
          });
        });
    }
  }

  reauthenticate = currentPassword => {
    var user = firebase.auth().currentUser;
    console.log(currentPassword);
    console.log(this.state.newPassword);
    var cred = firebase.auth.EmailAuthProvider.credential(
      user.email,
      currentPassword
    );
    console.log(cred);
    return user.reauthenticateAndRetrieveDataWithCredential(cred);
  };

  onChangePasswordPress = () => {
    this.reauthenticate(this.state.currentPassword)
      .then(() => {
        var user = firebase.auth().currentUser;
        user
          .updatePassword(this.state.newPassword)
          .then(() => {
            this.setState({
              notify: true,
              notifyMsg: "Password updated successfully!",
              processing: false
            });
          })
          .catch(error => {
            this.setState({
              notify: true,
              notifyMsg: error.message,
              processing: false
            });
          });
      })
      .catch(error => {
        this.setState({
          notify: true,
          notifyMsg: error.message,
          processing: false
        });
      });
  };

  render() {
    return (
      <div style={{ padding: "50px 200px" }}>
        {/*=============WELCOME USER HEADER=============*/}
        <div className="user-header">
          <Typography variant="display2" style={{ flex: 1 }}>
            {this.state.name}
            's Profile
            <Button
              style={{ float: "right" }}
              onClick={() => {
                this.props.history.push(`/user/home`);
              }}
            >
              <Back />
              Back
            </Button>
          </Typography>
        </div>
        <Divider />

        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={3000}
          message={this.state.notifyMsg}
        />
        <AppBar
          className="bar"
          data-aos="zoom-in-up"
          position="static"
          color="default"
        >
          <Tabs
            value={this.state.value}
            onChange={this.handleChangeTab}
            indicatorColor="primary"
            textColor="primary"
            fullWidth
          >
            <Tab label="User Info" />
            <Tab label="Change Email Section" />
            <Tab label="Change Password Section" />
          </Tabs>
        </AppBar>
        <SwipeableViews
          index={this.state.value}
          onChangeIndex={this.handleChangeIndex}
        >
          <TabContainer>
            <Card className="input-paper" data-aos="zoom-in-up">
              <CardContent>
                <TextField
                  id="name"
                  label="Full Name"
                  value={this.state.name}
                  fullWidth
                  className="push-down"
                  InputProps={{
                    readOnly: true
                  }}
                />

                <TextField
                  onChange={this.handleChange("address")}
                  id="address"
                  label="Full Address"
                  required
                  value={this.state.address}
                  fullWidth
                  className="push-down"
                  disabled={this.state.processing}
                />

                <Button
                  style={{
                    float: "right",
                    marginTop: 10
                  }}
                  variant="raised"
                  color="primary"
                  className="input-button"
                  onClick={this.submitNewAddress}
                >
                  Update Address
                </Button>
              </CardContent>
            </Card>
          </TabContainer>
          <TabContainer>{this.state.email}</TabContainer>
          <TabContainer>
            <Card className="input-paper" data-aos="zoom-in-up">
              <CardContent>
                <TextField
                  onChange={this.handleChange("currentPassword")}
                  id="currentPassword"
                  label="Current Password"
                  required
                  value={this.state.currentPassword}
                  fullWidth
                  type="password"
                  className="push-down"
                  disabled={this.state.processing}
                />

                <TextField
                  onChange={this.handleChange("newPassword")}
                  id="password"
                  label="New Password"
                  required
                  type="password"
                  value={this.state.newPassword}
                  fullWidth
                  className="push-down"
                  disabled={this.state.processing}
                />

                <Button
                  style={{
                    float: "right",
                    marginTop: 10
                  }}
                  variant="raised"
                  color="primary"
                  className="input-button"
                  onClick={this.onChangePasswordPress}
                >
                  Change Password
                </Button>
              </CardContent>
            </Card>
          </TabContainer>
        </SwipeableViews>
      </div>
    );
  }
}
export default UserProfile;
