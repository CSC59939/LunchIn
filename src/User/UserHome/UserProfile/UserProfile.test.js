import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import UserProfile from "./UserProfile";

describe("UserProfile Component", () => {
  describe("When provided with no props", () => {
    it("Should render correctly", () => {
      const userProfileInstance = shallow(
        <UserProfile />
      );
      expect(userProfileInstance).toMatchSnapshot();
    });
  });
});
