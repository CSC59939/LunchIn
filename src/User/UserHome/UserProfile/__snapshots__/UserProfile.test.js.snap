// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Component When provided with no props Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-header"
        >
          <WithStyles(Typography)
            style={
              Object {
                "flex": 1,
              }
            }
            variant="display2"
          >
            Profile
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "float": "right",
                }
              }
            >
              <pure(ArrowBackIcon) />
              Back
            </WithStyles(Button)>
          </WithStyles(Typography)>
        </div>,
        <WithStyles(Divider) />,
        <WithStyles(Snackbar)
          autoHideDuration={6000}
          message=""
          onClose={[Function]}
          open={false}
        />,
        <div
          className="user-profile-content"
        >
          <WithStyles(Card)
            className="input-paper"
            data-aos="zoom-in-up"
          >
            <WithStyles(CardContent)>
              <TextField
                InputProps={
                  Object {
                    "readOnly": true,
                  }
                }
                className="push-down"
                fullWidth={true}
                id="name"
                label="Full Name"
                required={false}
                select={false}
                value=""
                variant="standard"
              />
              <TextField
                className="push-down"
                disabled={false}
                fullWidth={true}
                id="address"
                label="Full Address"
                onChange={[Function]}
                required={true}
                select={false}
                value=""
                variant="standard"
              />
              <WithStyles(Button)
                className="input-button"
                color="primary"
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                    "marginTop": 10,
                  }
                }
                variant="raised"
              >
                Update
              </WithStyles(Button)>
            </WithStyles(CardContent)>
          </WithStyles(Card)>
        </div>,
      ],
      "style": Object {
        "padding": "50px 200px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Typography)
            style={
              Object {
                "flex": 1,
              }
            }
            variant="display2"
          >
            Profile
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "float": "right",
                }
              }
            >
              <pure(ArrowBackIcon) />
              Back
            </WithStyles(Button)>
          </WithStyles(Typography)>,
          "className": "user-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Profile",
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(ArrowBackIcon) />
                Back
              </WithStyles(Button)>,
            ],
            "style": Object {
              "flex": 1,
            },
            "variant": "display2",
          },
          "ref": null,
          "rendered": Array [
            "Profile",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <pure(ArrowBackIcon) />,
                  "Back",
                ],
                "onClick": [Function],
                "style": Object {
                  "float": "right",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Back",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoHideDuration": 6000,
          "message": "",
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Card)
            className="input-paper"
            data-aos="zoom-in-up"
          >
            <WithStyles(CardContent)>
              <TextField
                InputProps={
                  Object {
                    "readOnly": true,
                  }
                }
                className="push-down"
                fullWidth={true}
                id="name"
                label="Full Name"
                required={false}
                select={false}
                value=""
                variant="standard"
              />
              <TextField
                className="push-down"
                disabled={false}
                fullWidth={true}
                id="address"
                label="Full Address"
                onChange={[Function]}
                required={true}
                select={false}
                value=""
                variant="standard"
              />
              <WithStyles(Button)
                className="input-button"
                color="primary"
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                    "marginTop": 10,
                  }
                }
                variant="raised"
              >
                Update
              </WithStyles(Button)>
            </WithStyles(CardContent)>
          </WithStyles(Card)>,
          "className": "user-profile-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(CardContent)>
              <TextField
                InputProps={
                  Object {
                    "readOnly": true,
                  }
                }
                className="push-down"
                fullWidth={true}
                id="name"
                label="Full Name"
                required={false}
                select={false}
                value=""
                variant="standard"
              />
              <TextField
                className="push-down"
                disabled={false}
                fullWidth={true}
                id="address"
                label="Full Address"
                onChange={[Function]}
                required={true}
                select={false}
                value=""
                variant="standard"
              />
              <WithStyles(Button)
                className="input-button"
                color="primary"
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                    "marginTop": 10,
                  }
                }
                variant="raised"
              >
                Update
              </WithStyles(Button)>
            </WithStyles(CardContent)>,
            "className": "input-paper",
            "data-aos": "zoom-in-up",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TextField
                  InputProps={
                    Object {
                      "readOnly": true,
                    }
                  }
                  className="push-down"
                  fullWidth={true}
                  id="name"
                  label="Full Name"
                  required={false}
                  select={false}
                  value=""
                  variant="standard"
                />,
                <TextField
                  className="push-down"
                  disabled={false}
                  fullWidth={true}
                  id="address"
                  label="Full Address"
                  onChange={[Function]}
                  required={true}
                  select={false}
                  value=""
                  variant="standard"
                />,
                <WithStyles(Button)
                  className="input-button"
                  color="primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                      "marginTop": 10,
                    }
                  }
                  variant="raised"
                >
                  Update
                </WithStyles(Button)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "InputProps": Object {
                    "readOnly": true,
                  },
                  "className": "push-down",
                  "fullWidth": true,
                  "id": "name",
                  "label": "Full Name",
                  "required": false,
                  "select": false,
                  "value": "",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "push-down",
                  "disabled": false,
                  "fullWidth": true,
                  "id": "address",
                  "label": "Full Address",
                  "onChange": [Function],
                  "required": true,
                  "select": false,
                  "value": "",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Update",
                  "className": "input-button",
                  "color": "primary",
                  "onClick": [Function],
                  "style": Object {
                    "float": "right",
                    "marginTop": 10,
                  },
                  "variant": "raised",
                },
                "ref": null,
                "rendered": "Update",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-header"
          >
            <WithStyles(Typography)
              style={
                Object {
                  "flex": 1,
                }
              }
              variant="display2"
            >
              Profile
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(ArrowBackIcon) />
                Back
              </WithStyles(Button)>
            </WithStyles(Typography)>
          </div>,
          <WithStyles(Divider) />,
          <WithStyles(Snackbar)
            autoHideDuration={6000}
            message=""
            onClose={[Function]}
            open={false}
          />,
          <div
            className="user-profile-content"
          >
            <WithStyles(Card)
              className="input-paper"
              data-aos="zoom-in-up"
            >
              <WithStyles(CardContent)>
                <TextField
                  InputProps={
                    Object {
                      "readOnly": true,
                    }
                  }
                  className="push-down"
                  fullWidth={true}
                  id="name"
                  label="Full Name"
                  required={false}
                  select={false}
                  value=""
                  variant="standard"
                />
                <TextField
                  className="push-down"
                  disabled={false}
                  fullWidth={true}
                  id="address"
                  label="Full Address"
                  onChange={[Function]}
                  required={true}
                  select={false}
                  value=""
                  variant="standard"
                />
                <WithStyles(Button)
                  className="input-button"
                  color="primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                      "marginTop": 10,
                    }
                  }
                  variant="raised"
                >
                  Update
                </WithStyles(Button)>
              </WithStyles(CardContent)>
            </WithStyles(Card)>
          </div>,
        ],
        "style": Object {
          "padding": "50px 200px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Typography)
              style={
                Object {
                  "flex": 1,
                }
              }
              variant="display2"
            >
              Profile
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(ArrowBackIcon) />
                Back
              </WithStyles(Button)>
            </WithStyles(Typography)>,
            "className": "user-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Profile",
                <WithStyles(Button)
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                    }
                  }
                >
                  <pure(ArrowBackIcon) />
                  Back
                </WithStyles(Button)>,
              ],
              "style": Object {
                "flex": 1,
              },
              "variant": "display2",
            },
            "ref": null,
            "rendered": Array [
              "Profile",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <pure(ArrowBackIcon) />,
                    "Back",
                  ],
                  "onClick": [Function],
                  "style": Object {
                    "float": "right",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "Back",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoHideDuration": 6000,
            "message": "",
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Card)
              className="input-paper"
              data-aos="zoom-in-up"
            >
              <WithStyles(CardContent)>
                <TextField
                  InputProps={
                    Object {
                      "readOnly": true,
                    }
                  }
                  className="push-down"
                  fullWidth={true}
                  id="name"
                  label="Full Name"
                  required={false}
                  select={false}
                  value=""
                  variant="standard"
                />
                <TextField
                  className="push-down"
                  disabled={false}
                  fullWidth={true}
                  id="address"
                  label="Full Address"
                  onChange={[Function]}
                  required={true}
                  select={false}
                  value=""
                  variant="standard"
                />
                <WithStyles(Button)
                  className="input-button"
                  color="primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                      "marginTop": 10,
                    }
                  }
                  variant="raised"
                >
                  Update
                </WithStyles(Button)>
              </WithStyles(CardContent)>
            </WithStyles(Card)>,
            "className": "user-profile-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(CardContent)>
                <TextField
                  InputProps={
                    Object {
                      "readOnly": true,
                    }
                  }
                  className="push-down"
                  fullWidth={true}
                  id="name"
                  label="Full Name"
                  required={false}
                  select={false}
                  value=""
                  variant="standard"
                />
                <TextField
                  className="push-down"
                  disabled={false}
                  fullWidth={true}
                  id="address"
                  label="Full Address"
                  onChange={[Function]}
                  required={true}
                  select={false}
                  value=""
                  variant="standard"
                />
                <WithStyles(Button)
                  className="input-button"
                  color="primary"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                      "marginTop": 10,
                    }
                  }
                  variant="raised"
                >
                  Update
                </WithStyles(Button)>
              </WithStyles(CardContent)>,
              "className": "input-paper",
              "data-aos": "zoom-in-up",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <TextField
                    InputProps={
                      Object {
                        "readOnly": true,
                      }
                    }
                    className="push-down"
                    fullWidth={true}
                    id="name"
                    label="Full Name"
                    required={false}
                    select={false}
                    value=""
                    variant="standard"
                  />,
                  <TextField
                    className="push-down"
                    disabled={false}
                    fullWidth={true}
                    id="address"
                    label="Full Address"
                    onChange={[Function]}
                    required={true}
                    select={false}
                    value=""
                    variant="standard"
                  />,
                  <WithStyles(Button)
                    className="input-button"
                    color="primary"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "right",
                        "marginTop": 10,
                      }
                    }
                    variant="raised"
                  >
                    Update
                  </WithStyles(Button)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "InputProps": Object {
                      "readOnly": true,
                    },
                    "className": "push-down",
                    "fullWidth": true,
                    "id": "name",
                    "label": "Full Name",
                    "required": false,
                    "select": false,
                    "value": "",
                    "variant": "standard",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "push-down",
                    "disabled": false,
                    "fullWidth": true,
                    "id": "address",
                    "label": "Full Address",
                    "onChange": [Function],
                    "required": true,
                    "select": false,
                    "value": "",
                    "variant": "standard",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Update",
                    "className": "input-button",
                    "color": "primary",
                    "onClick": [Function],
                    "style": Object {
                      "float": "right",
                      "marginTop": 10,
                    },
                    "variant": "raised",
                  },
                  "ref": null,
                  "rendered": "Update",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
